### PR TITLE
feat: view binding and compose

### PR DESCRIPTION
**Pull request by FunkyMuse**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,3 +11,4 @@ keystore.properties
 fastlane/fastlane.json
 Gemfile
 Gemfile.lock
+/java_pid3915.hprof
```
Comment (tibbi) [inline .gitignore:14]: this isnt needed here, just delete it

File: app/src/main/kotlin/com/simplemobiletools/calculator/compose/extensions/BooleanPreviewParameterProvider.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package com.simplemobiletools.calculator.compose.extensions
+
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+
+class BooleanPreviewParameterProvider : PreviewParameterProvider<Boolean> {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calculator/compose/extensions/BooleanPreviewParameterProvider.kt:5]: what does this do?

File: app/src/main/kotlin/com/simplemobiletools/calculator/compose/settings/SettingsCheckBoxComponent.kt [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    checkboxColor: Color = MaterialTheme.colorScheme.primary
+) {
+    val interactionSource = remember { MutableInteractionSource() }
+    val indication = LocalIndication.current
+
+    Row(
+        verticalAlignment = Alignment.CenterVertically,
+        modifier = modifier
+            .fillMaxWidth()
+            .clickable(onClick = { onChange?.invoke(!initialValue) }, interactionSource = interactionSource, indication = indication)
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calculator/compose/settings/SettingsCheckBoxComponent.kt:49]: split such things into multiple lines please, there is too much going on

File: app/src/main/kotlin/com/simplemobiletools/calculator/compose/theme/Theme.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+) {
+    val view = LocalView.current
+    val context = LocalContext.current
+    val systemUiController = rememberSystemUiController()
+    val baseConfig = remember { context.config }
+
+    val colorScheme = if (!view.isInEditMode) {
+
+        val colorScheme = when {
+            theme is Theme.SystemDefaultMaterialYou && Build.VERSION.SDK_INT >= Build.VERSION_CODES.S -> {
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calculator/compose/theme/Theme.kt:33]: lets use our helper functions like isSPlus etc

File: app/src/main/kotlin/com/simplemobiletools/calculator/compose/theme/Theme.kt [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+                tertiary = theme.primaryColor,
+                onSurface = theme.textColor
+            )
+
+            else -> darkColorScheme
+        }
+
+        colorScheme
+
+    } else darkColorScheme
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/calculator/compose/theme/Theme.kt:61]: always use brackets at if/else

File: app/src/main/kotlin/com/simplemobiletools/calculator/compose/extensions/BooleanPreviewParameterProvider.kt (hunk 2 above)
Comment (FunkyMuse) [reply app/src/main/kotlin/com/simplemobiletools/calculator/compose/extensions/BooleanPreviewParameterProvider.kt:5]: https://developer.android.com/jetpack/compose/tooling/previews#preview-data

**Comment by tibbi** [comment]
that ripple is way too small, it should be the same as at other top menu icons

**Comment by FunkyMuse** [comment]
> that ripple is way too small, it should be the same as at other top menu icons

done

**Comment by tibbi** [comment]
its too big now, not consistent with menu items. Make it by like 4dp smaller

**Comment by FunkyMuse** [comment]
> its too big now, not consistent with menu items. Make it by like 4dp smaller

check now

**Comment by tibbi** [comment]
right, the size is good now, but it seems to be too lose to the left side, can we move it a bit to the right? Compare it with the About section.

**Comment by FunkyMuse** [comment]
> right, the size is good now, but it seems to be too lose to the left side, can we move it a bit to the right? Compare it with the About section.

sure, done

**Comment by tibbi** [comment]
alright thanks, it looks just fine to me now. But lets make the top bar reusable in a better way, I wouldnt like to copypaste the whole thing with all these dimensions in all activities and apps

**Pull request merged**